src/front-end/typescript/lib/framework/component/base.ts

```diff
@@ -126,6 +158,18 @@ export interface ComponentViewProps<State, Msg> {
 
 export type Dispatch<Msg> = (msg: Msg) => void;
 
+/**
+ * Typically used in the context of a view function,
+ * where an event such as a checkbox click needs to
+ * trigger a state change (via Msg).
+ *
+ * Dispatches a message from a child component to a
+ * parent.
+ *
+ * @param dispatch - ParentMsg, ChildMsg
+ * @param fn - typically an adt function
+ * @returns a Msg
+ */
```

Comment:
I recommend updating the following from:

```
Typically used in the context of a view function,
 * where an event such as a checkbox click needs to
 * trigger a state change (via Msg).
 *
 * Dispatches a message from a child component to a
 * parent.
 ```

to:
 
 ```
This function is used to create dispatch functions for child components
inside parent components. It is typically used inside parent components' view functions.
 ```
 
 Also, this:
 
 ```
 @param fn - typically an adt function
 ```
 
 should be:
 
 ```
 @param fn - a mapping function that converts a child component's Msg to a parent Msg so it can be handled by the parent component's update function.
 ```

Reply:
Also, this `@returns Dispatch<ChildMsg>`

Reply:
great recommendations. thank you...addressed in https://github.com/bcgov/digital_marketplace/pull/332/commits/730fc6e16be84e1472a5ec3879fcea37596f920f